File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int n = nums.size();
        for(auto i = 0; i < n-1 ; i++){
            if(i+2 < n && nums[i] == nums[i+2]){
                return nums[i];
            }
            if(nums[i] == nums[i+1]){
                return nums[i];
            }
        }
        return nums[0];
    }
};

File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.rs
impl Solution {
    pub fn repeated_n_times(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        for i in 0..n - 1{
            if i+2 < n && nums[i] == nums[i+2]{
                return nums[i];
            }
            if nums[i] == nums[i + 1]{
                return nums[i];
            }
        }
        nums[0]
    }
}